File: back/DevC/src/main/java/com/back/devc/domain/admin/dashboard/repository/AdminDashboardRepository.java
package com.back.devc.domain.admin.dashboard.repository;

import com.back.devc.domain.admin.dashboard.dto.DashboardResponseDto.ReportReasonCount;
import com.back.devc.domain.interaction.report.entity.ReportStatus;

import java.time.LocalDate;
import java.util.List;

public interface AdminDashboardRepository {

    Long countAllUsers();
    Long countAllPosts();

    Long countPendingReports();

    Long countTodayUsers(LocalDate today);
    Long countTodayPosts(LocalDate today);
    Long countTodayComments(LocalDate today);

    Long countTodayPostReports(LocalDate today);
    Long countTodayPostReportsByStatus(LocalDate today, ReportStatus status);
    List<ReportReasonCount> countTodayPostReportsByReason(LocalDate today);

    Long countTodayCommentReports(LocalDate today);
    Long countTodayCommentReportsByStatus(LocalDate today, ReportStatus status);
    List<ReportReasonCount> countTodayCommentReportsByReason(LocalDate today);
}

File: back/DevC/src/main/java/com/back/devc/domain/admin/dashboard/repository/AdminDashboardRepository.kt
package com.back.devc.domain.admin.dashboard.repository

import com.back.devc.domain.admin.dashboard.dto.DashboardResponseDto.ReportReasonCount
import com.back.devc.domain.interaction.report.entity.ReportStatus
import java.time.LocalDate

interface AdminDashboardRepository {
    fun countAllUsers(): Long

    fun countAllPosts(): Long

    fun countPendingReports(): Long

    fun countTodayUsers(today: LocalDate): Long

    fun countTodayPosts(today: LocalDate): Long

    fun countTodayComments(today: LocalDate): Long

    fun countTodayPostReports(today: LocalDate): Long

    fun countTodayPostReportsByStatus(today: LocalDate, status: ReportStatus): Long

    fun countTodayPostReportsByReason(today: LocalDate): List<ReportReasonCount>

    fun countTodayCommentReports(today: LocalDate): Long

    fun countTodayCommentReportsByStatus(today: LocalDate, status: ReportStatus): Long

    fun countTodayCommentReportsByReason(today: LocalDate): List<ReportReasonCount>
}
